tests/single-instance.tftest.hcl

```diff
@@ -0,0 +1,130 @@
+variables {
+  root_modules_path = "./tests/fixtures/single-instance"
+  github_enterprise = {
+    namespace = "masterpointio"
+  }
+  repository = "terraform-spacelift-automation"
+  all_root_modules_enabled = true
+  aws_integration_enabled = false
+  before_init = [
+    "echo 'Hello'"
+  ]
+
+  root_module_structure = "SingleInstance"
+}
+
+# Test that the root module fileset is created correctly
+run "test_single_instance_root_module_fileset_collects_all_root_modules" {
+  command = plan
+
+  assert {
+    condition = contains(local._all_root_modules, "root-module-a")
+    error_message = "Root module fileset was not created correctly: ${jsonencode(local._all_root_modules)}"
+  }
+}
+
+# Test that the stack names are created correctly
+run "test_single_instance_stacks_include_expected" {
+  command = plan
+
+  assert {
+    condition = contains(local.stacks, "root-module-a")
+    error_message = "Stack names were not created correctly: ${jsonencode(local.stacks)}"
+  }
+}
+
+run "test_single_instance_stacks_only_include_default_stack" {
+  command = plan
+
+  assert {
+    condition = length(local._root_module_yaml_decoded["root-module-a"]) == 1 && local._root_module_yaml_decoded["root-module-a"]["default"] != null
+    error_message = "_root_module_yaml_decoded is not a single instance: ${jsonencode(local._root_module_yaml_decoded)}"
+  }
+}
+
+run "test_single_instance_stack_configs_stack_name_is_correct" {
+  command = plan
+
+  assert {
+    condition = length(local._root_module_stack_configs) == 1 && local._root_module_stack_configs["root-module-a"] != null
+    error_message = "_root_module_stack_configs is not expected structure: ${jsonencode(local._root_module_stack_configs)}"
+  }
+}
+
+run "test_single_instance_stack_configs_use_default_tf_workspace" {
+  command = plan
+
+  assert {
+    condition = local._root_module_stack_configs["root-module-a"].terraform_workspace == "default"
+    error_message = "terraform_workspace is not set to default: ${jsonencode(local._root_module_stack_configs)}"
+  }
+}
+
+run "test_single_instance_stack_configs_project_root_is_correct" {
+  command = plan
+
+  assert {
+    condition = local._root_module_stack_configs["root-module-a"].project_root == "${var.root_modules_path}/root-module-a"
+    error_message = "project_root is not correct for root-module-a: ${jsonencode(local._root_module_stack_configs)}"
+  }
+}
+
```

Comment:
These tests are awesome! But yeah a way to specify the `project_root` is needed.

Reply:
The tests will help with implementing!!!